src/api/app/models/concerns/workflow_run_gitlab_payload.rb

```diff
@@ -0,0 +1,53 @@
+# Methods to know which webhook we are dealing with based on the request_payload attribute
+class WorkflowRunGitlabPayload
+  extend ActiveSupport::Concern
+
+  ALLOWED_MERGE_REQUEST_ACTIONS = %w[close merge open reopen update].freeze
+
+  private
+
+  def gitlab_repository_name
+    payload.dig('project', 'path_with_namespace')&.split('/')&.last
+  end
+
+  def gitlab_repository_owner
+    payload.dig('project', 'path_with_namespace')&.split('/')&.first
+  end
+
+  def gitlab_hook_action
+    payload.dig('object_attributes', 'action')
+  end
+
+  def api_endpoint
```

Comment:
this is duplicated from `WorkflowRunPayload` and should not be here

Reply:
Removed the duplication